### ISSUE TITLE
Question networkidle2

### ISSUE THREAD

**Issue by ctaity** [ulixee/secret-agent#345]
Exists some event to wait for the page to load completely like networkidle2 in puppeteer ? Thanks.

**Comment by ctaity** [comment]
Yes but not always the element is loaded or created, for example, an element created by an ajax call or something like that. For example tiktok search page has 4 versions, and secret agent wait ccs3 selectors in serial manner if i wait 5 seconds , takes 20 seconds to load de the page. With network idle, when de page is loaded i use the innerhtml and search in sync way the selectors, with this way it tooks 5 seconds.

The function i made to wait a selector is this: 

`const waitForSelector = async (
  agent: Agent,
  selector: string,
  visible: boolean,
  timeout: number,
): Promise<any> => {
  try {
    console.log('waiting for element:' + selector);
    await agent.waitForElement(agent.document.querySelector(selector), {
      timeoutMs: timeout,
      waitForVisible: visible,
    });
    return agent.document.querySelector(selector);
  } catch (e) {
    console.log('TIMEOUT FOR SELECTOR:' + selector);
    return null;
  }
};`

**Comment by blakebyrnes** [comment]
This makes sense. We're just wrapping up a feature in Hero (next iteration of SecretAgent) that I think should help with this scenario (https://github.com/ulixee/hero/pull/26). We're not quite ready for people to switch over, but might give you a taste of what's coming!
